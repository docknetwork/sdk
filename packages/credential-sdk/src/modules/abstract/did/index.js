export { default as AbstractDIDModule } from './module';
export * from './errors';
