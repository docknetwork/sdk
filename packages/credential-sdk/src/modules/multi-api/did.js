import { AbstractDIDModule } from '../abstract';
import { DIDDocument, NamespaceDid } from '../../types';
import { injectModuleRouter } from './common';

export default class MultiApiDIDModule extends injectModuleRouter(
  AbstractDIDModule,
) {
  /**
   *
   * @param {DIDDocument} didDocument
   * @param {DidKeypair} didKeypair
   * @returns {Promise<void>}
   */
  async createDocument(didDocument, didKeypair, params) {
    const document = DIDDocument.from(didDocument);

    return await this.moduleById(document.id).createDocument(
      document,
      didKeypair,
      params,
    );
  }

  /**
   *
   * @param {DIDDocument} didDocument
   * @param {DidKeypair} didKeypair
   * @returns {Promise<void>}
   */
  async updateDocument(didDocument, didKeypair, params) {
    const document = DIDDocument.from(didDocument);

    return await this.moduleById(document.id).updateDocument(
      document,
      didKeypair,
      params,
    );
  }

  /**
   *
   * @param {*} did
   * @param {DidKeypair} didKeypair
   * @returns {Promise<void>}
   */
  async removeDocument(did, didKeypair, params) {
    const id = NamespaceDid.from(did);

    return await this.moduleById(id).removeDocument(id, didKeypair, params);
  }

  /**
   *
   * @param {*} did
   * @returns {Promise<DIDDocument>}
   */
  async getDocument(did) {
    const id = NamespaceDid.from(did);

    return await this.moduleById(id).getDocument(id);
  }

  /**
   *
   * @param {DIDDocument} didDocument
   * @param {DidKeypair} didKeypair
   * @returns {Promise<void>}
   */
  async createDocumentTx(didDocument, didKeypair) {
    const document = DIDDocument.from(didDocument);

    return await this.moduleById(document.id).createDocumentTx(
      document,
      didKeypair,
    );
  }

  /**
   *
   * @param {DIDDocument} didDocument
   * @param {DidKeypair} didKeypair
   * @returns {Promise<void>}
   */
  async updateDocumentTx(didDocument, didKeypair) {
    const document = DIDDocument.from(didDocument);

    return await this.moduleById(document.id).updateDocumentTx(
      document,
      didKeypair,
    );
  }

  /**
   *
   * @param {*} did
   * @param {DidKeypair} didKeypair
   * @returns {Promise<void>}
   */
  async removeDocumentTx(did, didKeypair) {
    const id = NamespaceDid.from(did);

    return await this.moduleById(id).removeDocumentTx(id, didKeypair);
  }
}
