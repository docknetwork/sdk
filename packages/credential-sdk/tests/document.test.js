import {
  VerificationRelationship,
  DidKey,
  CheqdTestnetDid,
} from "../src/types/did";
import { DIDDocument } from "../src/types/did/document";
import { PublicKeyEd25519, BBSPublicKey, BBSPublicKeyValue } from "../src/types";

const RANDOM_PKS = [
  "0xa1aa6a2058dd190e284a64e72adaf4e16a9ae9fbf0673d7575924e6aca3b21dc",
  "0xed28a616e8f8cfd2232b62dac57281b24fc0d538150f8f5d91a0b6c3255cc2c9",
  "0x3f10af3f08a442251510f61ef2395b6419f30f5eaac2114864f72731452a71ec",
  "0xc9b4c22c2f5738b0adaf605a8233cdc2451c5ece045c25a3bc2b4c840fe9b11d",
  "0x86839d44e6f15cd21ede6e4db64c60b57cb4c5f6030212f48468090c1ed823ed",
  "0x91253c4b3401eade456953934325153c908444b8ea2705f8ee71e600bee6b0bc",
  "0xddf88ee3d89f5890ab0b14d37b9189e62cf88baf375557f917994edaec8b296d",
   "0x9d1aa9f5f7990737f670accdbdb8808ab4cbf256e6aaac73cba8bd89aa04befb",
].map((bytes) => new PublicKeyEd25519(bytes));

const RANDOM_OFFCHAIN_PKS = [
  "0xacfa7cba73ee6d09d280c157b5e25c4354da462103568f3c2036f714de1a8b18",
  "0x853a2e866edb66d965326c019f039bc9c3b83d02ee5aa316fbe41edf55c45e0f",
  "0x97349a1977e401a9472587c2f81473fb20967c0f24b6dd1d1c8fb2a3833f6374",
    "0xc9b4c22c2f5738b0adaf605a8233cdc2451c5ece045c25a3bc2b4c840fe9b11d",
].map(bytes => new BBSPublicKey(new BBSPublicKeyValue(bytes)));

const catchErrorWith = (fn, message) => {
  try {
    return fn();
  } catch(err) {
    err.message = `${message}:\n${err.message}`;

    throw err;
  }
}

const checkDoc = doc => {
  catchErrorWith(() => expect(doc).toMatchSnapshot(), 'Failed snapshot check');
  const jsonDoc = catchErrorWith(() => doc.toJSON(), 'Failed `toJSON` conversion');
  const newDoc = catchErrorWith(() => doc.constructor.from(jsonDoc), 'Failed `fromJSON` conversion');
  catchErrorWith(() => expect(newDoc.toJSON()).toEqual(jsonDoc), 'Failed equality check');
}

const checkDocs = (...docs) => {
  let idx = 0;
  for (const doc of docs) {
    catchErrorWith(() => {
      checkDoc(doc);
      idx++;
    }, `Error for docs[${idx}]`)
  }
}

describe("DID document workflow", () => {
  const AUTH = new VerificationRelationship().setAuthentication();
  const ASSERT = new VerificationRelationship().setAssertion();
  const CAP_INV = new VerificationRelationship().setCapabilityInvocation();
  const KEY_AGR = new VerificationRelationship().setKeyAgreement();

  test(`\`DIDDocument.from\` works`, () => {
    const doc = {
      "@context": ["test"],
      id: "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk",
      controller: ["did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk"],
      verificationMethod: [
        {
          controller:
            "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk",
          id: "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk#keys-3",
          publicKeyBase58: "EaNp37NGpe8KViTKxAdknyvUP44doH1fHnFpYqEcfUjE",
          type: "Ed25519VerificationKey2018",
        },
      ],
    };

    expect(DIDDocument.from(doc)).toMatchSnapshot();
  });

  test("`DIDDocument`", () => {
    const doc1 =       DIDDocument.create(
      "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk",
      [],
      [],
      []
    );

    const doc2 =       DIDDocument.create(
      "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk",
      [
        new DidKey(RANDOM_PKS[0], AUTH),
        new DidKey(RANDOM_OFFCHAIN_PKS[0]),
        new DidKey(RANDOM_PKS[1], ASSERT),
        new DidKey(RANDOM_PKS[2], CAP_INV),
        new DidKey(RANDOM_OFFCHAIN_PKS[1]),
        new DidKey(RANDOM_PKS[3], AUTH | ASSERT),
        new DidKey(RANDOM_PKS[4], AUTH | CAP_INV),
        new DidKey(RANDOM_PKS[5], ASSERT | CAP_INV),
        new DidKey(RANDOM_PKS[6], AUTH | ASSERT | CAP_INV),
        new DidKey(RANDOM_PKS[7], KEY_AGR),
            new DidKey(RANDOM_OFFCHAIN_PKS[2]),
            new DidKey(RANDOM_OFFCHAIN_PKS[3]),
      ],
      [
        "did:key:z6MktvqCyLxTsXUH1tUZncNdVeEZ7hNh7npPRbUU27GTrYb8",
        "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk",
      ],
      []
    );

    const doc3 =       DIDDocument.create(
      "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk",
      [],
      [],
      [],
      { context: ["overriden"] }
    );

    checkDocs(doc1, doc2, doc3);
  });

  test("`CheqdDIDDocument`", () => {
    const cheqdDid = CheqdTestnetDid.from(
      "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54"
    );

    const doc1 = DIDDocument.create(
      "did:cheqd:testnet:6ec66297-f55e-4744-b8cd-bd043bc45767",
      [],
      [],
      []
    ).toCheqd('0557d7da-8dea-45cb-8f30-7370b59eab71');

    const doc2 = DIDDocument.create(cheqdDid, [new DidKey(RANDOM_PKS[0]), new DidKey(RANDOM_OFFCHAIN_PKS[0])], [cheqdDid], []).toCheqd('28edc043-872b-4ab7-9cb8-b01d1bd677c5');

    const doc3 = DIDDocument.create(
      "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54",
      [
        new DidKey(RANDOM_PKS[0], AUTH),
        new DidKey(RANDOM_OFFCHAIN_PKS[0]),
        new DidKey(RANDOM_PKS[1], ASSERT),
        new DidKey(RANDOM_PKS[2], CAP_INV),
        new DidKey(RANDOM_OFFCHAIN_PKS[1]),
        new DidKey(RANDOM_PKS[3], AUTH | ASSERT),
        new DidKey(RANDOM_PKS[4], AUTH | CAP_INV),
        new DidKey(RANDOM_PKS[5], ASSERT | CAP_INV),
        new DidKey(RANDOM_PKS[6], AUTH | ASSERT | CAP_INV),
        new DidKey(RANDOM_PKS[7], KEY_AGR),
        new DidKey(RANDOM_OFFCHAIN_PKS[2]),
        new DidKey(RANDOM_OFFCHAIN_PKS[3]),
      ],
      [
        "did:cheqd:mainnet:f1749383-d9dd-479f-82aa-e52fe8f59c54",
        "did:cheqd:testnet:df3ba823-c1e2-46e0-bb54-056f9c506e90",
      ],
      []
    ).toCheqd('47b4d97b-1ebe-4d81-a3ed-12eb63baaeb9');

    checkDocs(doc1, doc2, doc3)
  })
});
