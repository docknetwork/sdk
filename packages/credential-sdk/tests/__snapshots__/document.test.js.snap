// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DID document workflow \`CheqdDIDDocument\` 1`] = `
{
  "alsoKnownAs": [],
  "assertionMethod": [],
  "authentication": [],
  "capabilityDelegation": [],
  "capabilityInvocation": [],
  "context": [
    "https://www.w3.org/ns/did/v1",
  ],
  "controller": [],
  "id": "did:cheqd:testnet:6ec66297-f55e-4744-b8cd-bd043bc45767",
  "keyAgreement": [],
  "service": [],
  "verificationMethod": [],
  "versionId": "0557d7da-8dea-45cb-8f30-7370b59eab71",
}
`;

exports[`DID document workflow \`CheqdDIDDocument\` 2`] = `
{
  "alsoKnownAs": [],
  "assertionMethod": [
    "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54#keys-1",
    "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54#keys-2=BBXr5GktzTvvVhRsbmYdJw2Tw3LHbkL8GBewX7KoKTZC8cVhbtbm23jCSmkiqnLEs2vwJDQWFFb5FvNJhDuCHPaZkVobYByLv57KQYVriE7CwrzjXPcp61RVq6sChaE3hQ7SvDVyF55zrZ7xZTyXYKDMLvhYKmHwL8kJ1o9KAq5ZNBFJFSqhCJvejjkPXHg5YXPHAvFKS",
  ],
  "authentication": [
    "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54#keys-1",
  ],
  "capabilityDelegation": [],
  "capabilityInvocation": [
    "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54#keys-1",
  ],
  "context": [
    "https://www.w3.org/ns/did/v1",
  ],
  "controller": [
    "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54",
  ],
  "id": "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54",
  "keyAgreement": [],
  "service": [],
  "verificationMethod": [
    {
      "controller": "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54",
      "id": "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54#keys-1",
      "verificationMaterial": "Bt5Mafvpcqou1pBF8SnW1BJMvpkSzZnPEgWunTUFwHZm",
      "verificationMethodType": "Ed25519VerificationKey2018",
    },
  ],
  "versionId": "28edc043-872b-4ab7-9cb8-b01d1bd677c5",
}
`;

exports[`DID document workflow \`CheqdDIDDocument\` 3`] = `
{
  "alsoKnownAs": [],
  "assertionMethod": [
    "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54#keys-3",
    "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54#keys-6",
    "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54#keys-8",
    "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54#keys-9",
    "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54#keys-2=BBXr5GktzTvvVhRsbmYdJw2Tw3LHbkL8GBewX7KoKTZC8cVhbtbm23jCSmkiqnLEs2vwJDQWFFb5FvNJhDuCHPaZkVobYByLv57KQYVriE7CwrzjXPcp61RVq6sChaE3hQ7SvDVyF55zrZ7xZTyXYKDMLvhYKmHwL8kJ1o9KAq5ZNBFJFSqhCJvejjkPXHg5YXPHAvFKS",
    "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54#keys-5=BBXr5GktzTvvVhRsbmYdJw2Tw3G5BoeByaj9iFtGToMCQHXwjsjRvLUpz94VaiJVAYrs4uyKBmBoMN89tbo4dzX5nizBn6MjwbZXKHEKd2ZBzXaVg4TTdw6VHXT9nWYs5PXBEZBWEncp5mZ4pUMwhLk6XQUgSXgYHN9PHik8vszRLLrN1TkWSKn1D2VAm2aGmHyu6Mtnp",
    "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54#keys-11=BBXr5GktzTvvVhRsbmYdJw2Tw3GBAjToki5h3osi8WMS3RF4QLNS1D3atcFAHFFFFbXfETadaeApWKZsx5jF1WTtngunVo67bFhrd3KxFWtoYUoxtxNABmZpYPTP6HWeadskCqZFZeKVmCu2uu9TtuNjgBA5shwVHcc5cbmEsYavv4nq8oenCzMwjQ4cCxDE2KNHyFXL8",
    "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54#keys-12=BBXr5GktzTvvVhRsbmYdJw2Tw3LUVjBmqG8HFzgouT6CmR8j6gN1hRg9wg6eBBNoffP4D42hRbzknBxiTSxZAPri5yrWHGKwohGZtirDvLDunXcVA1a4ATw4WdLsfJxK4ayTSqtBhWRJJ8Gcp6yKZBUY96K7242JRcBWP4gzgQsCtdf1AUDEWc5CKsGZ5LfSuTgfKrUQt",
  ],
  "authentication": [
    "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54#keys-1",
    "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54#keys-6",
    "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54#keys-7",
    "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54#keys-9",
  ],
  "capabilityDelegation": [],
  "capabilityInvocation": [
    "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54#keys-4",
    "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54#keys-7",
    "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54#keys-8",
    "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54#keys-9",
  ],
  "context": [
    "https://www.w3.org/ns/did/v1",
  ],
  "controller": [
    "did:cheqd:mainnet:f1749383-d9dd-479f-82aa-e52fe8f59c54",
    "did:cheqd:testnet:df3ba823-c1e2-46e0-bb54-056f9c506e90",
  ],
  "id": "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54",
  "keyAgreement": [
    "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54#keys-10",
  ],
  "service": [],
  "verificationMethod": [
    {
      "controller": "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54",
      "id": "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54#keys-1",
      "verificationMaterial": "Bt5Mafvpcqou1pBF8SnW1BJMvpkSzZnPEgWunTUFwHZm",
      "verificationMethodType": "Ed25519VerificationKey2018",
    },
    {
      "controller": "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54",
      "id": "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54#keys-3",
      "verificationMaterial": "GxmaKA52SZGQtZS3Usa6Az48gky2AZRKzRKxsXM6ystc",
      "verificationMethodType": "Ed25519VerificationKey2018",
    },
    {
      "controller": "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54",
      "id": "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54#keys-4",
      "verificationMaterial": "5FBSa4xAs54osiMs75JcepL695v1XhccQmQePBZQaad9",
      "verificationMethodType": "Ed25519VerificationKey2018",
    },
    {
      "controller": "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54",
      "id": "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54#keys-6",
      "verificationMaterial": "EaNp37NGpe8KViTKxAdknyvUP44doH1fHnFpYqEcfUjE",
      "verificationMethodType": "Ed25519VerificationKey2018",
    },
    {
      "controller": "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54",
      "id": "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54#keys-7",
      "verificationMaterial": "A462rFkYENhY6kf9zLVPg6qvtQhXJ6RuL8CNdHTWzNqa",
      "verificationMethodType": "Ed25519VerificationKey2018",
    },
    {
      "controller": "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54",
      "id": "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54#keys-8",
      "verificationMaterial": "Amb3kf4if5Av1sz6UdYtEw2iy6DzufRAtch6hiXEWHcw",
      "verificationMethodType": "Ed25519VerificationKey2018",
    },
    {
      "controller": "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54",
      "id": "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54#keys-9",
      "verificationMaterial": "FwUvnTThdnNPv23Rbq2SXtxxYAT21Atvdd1cZBHPYue4",
      "verificationMethodType": "Ed25519VerificationKey2018",
    },
    {
      "controller": "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54",
      "id": "did:cheqd:testnet:f1749383-d9dd-479f-82aa-e52fe8f59c54#keys-10",
      "verificationMaterial": "BaGbEav6KzvWhgBWPyZK2vWX5EDDFc4a8rfUwAZ3x8Jn",
      "verificationMethodType": "Ed25519VerificationKey2018",
    },
  ],
  "versionId": "47b4d97b-1ebe-4d81-a3ed-12eb63baaeb9",
}
`;

exports[`DID document workflow \`DIDDocument.from\` works 1`] = `
{
  "@context": [
    "test",
  ],
  "alsoKnownAs": null,
  "assertionMethod": null,
  "authentication": null,
  "capabilityDelegation": null,
  "capabilityInvocation": null,
  "controller": [
    "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk",
  ],
  "https://rdf.dock.io/alpha/2021#attestsDocumentContents": null,
  "id": "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk",
  "keyAgreement": null,
  "service": null,
  "verificationMethod": [
    {
      "controller": "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk",
      "id": "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk#keys-3",
      "publicKeyBase58": "EaNp37NGpe8KViTKxAdknyvUP44doH1fHnFpYqEcfUjE",
      "publicKeyBase64": null,
      "publicKeyHex": null,
      "publicKeyJwk": null,
      "type": "Ed25519VerificationKey2018",
    },
  ],
}
`;

exports[`DID document workflow \`DIDDocument\` 1`] = `
{
  "@context": [
    "https://www.w3.org/ns/did/v1",
  ],
  "alsoKnownAs": [],
  "assertionMethod": [],
  "authentication": [],
  "capabilityDelegation": [],
  "capabilityInvocation": [],
  "controller": [],
  "https://rdf.dock.io/alpha/2021#attestsDocumentContents": null,
  "id": "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk",
  "keyAgreement": [],
  "service": [],
  "verificationMethod": [],
}
`;

exports[`DID document workflow \`DIDDocument\` 2`] = `
{
  "@context": [
    "https://www.w3.org/ns/did/v1",
  ],
  "alsoKnownAs": [],
  "assertionMethod": [
    "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk#keys-2",
    "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk#keys-3",
    "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk#keys-5",
    "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk#keys-6",
    "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk#keys-8",
    "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk#keys-9",
    "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk#keys-11",
    "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk#keys-12",
  ],
  "authentication": [
    "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk#keys-1",
    "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk#keys-6",
    "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk#keys-7",
    "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk#keys-9",
  ],
  "capabilityDelegation": [],
  "capabilityInvocation": [
    "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk#keys-4",
    "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk#keys-7",
    "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk#keys-8",
    "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk#keys-9",
  ],
  "controller": [
    "did:key:z6MktvqCyLxTsXUH1tUZncNdVeEZ7hNh7npPRbUU27GTrYb8",
    "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk",
  ],
  "https://rdf.dock.io/alpha/2021#attestsDocumentContents": null,
  "id": "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk",
  "keyAgreement": [
    "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk#keys-10",
  ],
  "service": [],
  "verificationMethod": [
    {
      "controller": "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk",
      "id": "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk#keys-1",
      "publicKeyBase58": "Bt5Mafvpcqou1pBF8SnW1BJMvpkSzZnPEgWunTUFwHZm",
      "publicKeyBase64": null,
      "publicKeyHex": null,
      "publicKeyJwk": null,
      "type": "Ed25519VerificationKey2018",
    },
    {
      "controller": "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk",
      "id": "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk#keys-2",
      "publicKeyBase58": "CeEf4Q1kvLo66eSbTSujuasf2mv1rTQEpK8rnuyqMxpP",
      "publicKeyBase64": null,
      "publicKeyHex": null,
      "publicKeyJwk": null,
      "type": "Bls12381BBSVerificationKeyDock2023",
    },
    {
      "controller": "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk",
      "id": "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk#keys-3",
      "publicKeyBase58": "GxmaKA52SZGQtZS3Usa6Az48gky2AZRKzRKxsXM6ystc",
      "publicKeyBase64": null,
      "publicKeyHex": null,
      "publicKeyJwk": null,
      "type": "Ed25519VerificationKey2018",
    },
    {
      "controller": "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk",
      "id": "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk#keys-4",
      "publicKeyBase58": "5FBSa4xAs54osiMs75JcepL695v1XhccQmQePBZQaad9",
      "publicKeyBase64": null,
      "publicKeyHex": null,
      "publicKeyJwk": null,
      "type": "Ed25519VerificationKey2018",
    },
    {
      "controller": "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk",
      "id": "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk#keys-5",
      "publicKeyBase58": "9y4gScRXDQJahd1gKbf1pYf9mwcRKnUqzstR3Bn7ydYz",
      "publicKeyBase64": null,
      "publicKeyHex": null,
      "publicKeyJwk": null,
      "type": "Bls12381BBSVerificationKeyDock2023",
    },
    {
      "controller": "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk",
      "id": "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk#keys-6",
      "publicKeyBase58": "EaNp37NGpe8KViTKxAdknyvUP44doH1fHnFpYqEcfUjE",
      "publicKeyBase64": null,
      "publicKeyHex": null,
      "publicKeyJwk": null,
      "type": "Ed25519VerificationKey2018",
    },
    {
      "controller": "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk",
      "id": "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk#keys-7",
      "publicKeyBase58": "A462rFkYENhY6kf9zLVPg6qvtQhXJ6RuL8CNdHTWzNqa",
      "publicKeyBase64": null,
      "publicKeyHex": null,
      "publicKeyJwk": null,
      "type": "Ed25519VerificationKey2018",
    },
    {
      "controller": "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk",
      "id": "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk#keys-8",
      "publicKeyBase58": "Amb3kf4if5Av1sz6UdYtEw2iy6DzufRAtch6hiXEWHcw",
      "publicKeyBase64": null,
      "publicKeyHex": null,
      "publicKeyJwk": null,
      "type": "Ed25519VerificationKey2018",
    },
    {
      "controller": "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk",
      "id": "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk#keys-9",
      "publicKeyBase58": "FwUvnTThdnNPv23Rbq2SXtxxYAT21Atvdd1cZBHPYue4",
      "publicKeyBase64": null,
      "publicKeyHex": null,
      "publicKeyJwk": null,
      "type": "Ed25519VerificationKey2018",
    },
    {
      "controller": "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk",
      "id": "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk#keys-10",
      "publicKeyBase58": "BaGbEav6KzvWhgBWPyZK2vWX5EDDFc4a8rfUwAZ3x8Jn",
      "publicKeyBase64": null,
      "publicKeyHex": null,
      "publicKeyJwk": null,
      "type": "Ed25519VerificationKey2018",
    },
    {
      "controller": "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk",
      "id": "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk#keys-11",
      "publicKeyBase58": "BBF5rtFSaWt4s4PxMEswkothcPJNLhdsQpggZyUozSdZ",
      "publicKeyBase64": null,
      "publicKeyHex": null,
      "publicKeyJwk": null,
      "type": "Bls12381BBSVerificationKeyDock2023",
    },
    {
      "controller": "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk",
      "id": "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk#keys-12",
      "publicKeyBase58": "EaNp37NGpe8KViTKxAdknyvUP44doH1fHnFpYqEcfUjE",
      "publicKeyBase64": null,
      "publicKeyHex": null,
      "publicKeyJwk": null,
      "type": "Bls12381BBSVerificationKeyDock2023",
    },
  ],
}
`;

exports[`DID document workflow \`DIDDocument\` 3`] = `
{
  "@context": [
    "overriden",
  ],
  "alsoKnownAs": [],
  "assertionMethod": [],
  "authentication": [],
  "capabilityDelegation": [],
  "capabilityInvocation": [],
  "controller": [],
  "https://rdf.dock.io/alpha/2021#attestsDocumentContents": null,
  "id": "did:dock:5DEHasvC9G3eVF3qCsN2VQvEbHYdQtsv74ozZ1ngQQj39Luk",
  "keyAgreement": [],
  "service": [],
  "verificationMethod": [],
}
`;
